ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/context/KsqlRestServiceContextFactory.java

```diff
@@ -35,43 +31,38 @@
  */
 public class KsqlRestServiceContextFactory implements Factory<ServiceContext> {
   private static KsqlConfig ksqlConfig;
+  private static KsqlSecurityExtension securityExtension;
 
-  public static void configure(final KsqlConfig ksqlConfig) {
+  public static void configure(
+      final KsqlConfig ksqlConfig,
+      final KsqlSecurityExtension securityExtension
+  ) {
     KsqlRestServiceContextFactory.ksqlConfig = Objects.requireNonNull(ksqlConfig, "ksqlConfig");
+    KsqlRestServiceContextFactory.securityExtension
+        = Objects.requireNonNull(securityExtension, "securityExtension");
   }
 
-  private final Optional<KsqlRestContext> ksqlRestContext;
+  private final SecurityContext securityContext;
 
   @Inject
-  public KsqlRestServiceContextFactory(final ContainerRequestContext requestContext) {
-    this.ksqlRestContext = KsqlRestContext.get(requestContext);
+  public KsqlRestServiceContextFactory(final SecurityContext securityContext) {
+    this.securityContext = securityContext;
   }
 
   @Override
   public ServiceContext provide() {
-    if (ksqlRestContext.isPresent()) {
-      return DefaultServiceContext.create(
-          ksqlConfig,
-          getKafkaClientSupplier(),
-          getSchemaRegistryClientSupplier()
-      );
+    if (!securityExtension.getAuthorizationProvider().isPresent()) {
```

Comment:
This looks like a bug: Why are we checking for the authorization provider here? Shouldn't we check for the user context provider instead? If this is indeed a bug, we need a unit test to cover this.

Reply:
Aa good catch, I didn't see it. I changed to to check for the UserContextProvider.
I added a couple of tests, however, I am planning to remove this `KsqlRestServiceContextFactory` class in a follow-up PR, and have `UserServiceContextFactory` that have a similar logic.